### PR TITLE
[Tests] Check if st2 services are listening on respective ports

### PR DESCRIPTION
**Pull request by armab**
Add real check if `st2api`, `st2auth`, `mistral` are reachable on `9100`, `9101`, `8989`

Guide:
http://serverspec.org/resource_types.html

-----
First we should include `netstat` binary, required for these checks, which is not available under minimal Docker.

@dennybaa can you add `net-tools` for all test Docker images & publish them, see: https://github.com/StackStorm/st2-dockerfiles/pull/7 ?

### CODE REVIEW

**Comment by armab** [comment]
Tests should fail: https://circleci.com/gh/armab/st2-packages/432 until we add `netstat` for all Docker images here: https://quay.io/repository/dennybaa/droneunit.

P.S. I like those Rspec tests, finally had chance to touch them :cake:

File: rake/spec/default/60-st2_all-services-ok_spec.rb [hunk 1]
```diff
@@ -9,17 +9,29 @@
     subject { host(spec[:rabbitmqhost]) }
     it { is_expected.to be_reachable }
   end
+  describe 'rabbitmq' do
+    subject { host(spec[:rabbitmqhost]) }
+    it { should be_reachable.with( :port => 5672, :timeout => 1 ) }
```
Comment (dennybaa) [inline rake/spec/default/60-st2_all-services-ok_spec.rb]: examples can have multiple tests, so above should be merge together/

either 
```ruby
   describe 'mongodb' do
      subject { host(spec[:mongodbhost]) }
      it { is_expected.to be_reachable }
      it { is_expected.to be_reachable.with :port => 27017, :timeout => 1 }
    end
```

or 
```ruby
   describe 'mongodb' do
      subject { host(spec[:mongodbhost]) }
      it { is_expected.to be_reachable.with :port => 27017, :timeout => 1 }
    end
```

PS. use defualt syntax (not should syntax)

**Comment by dennybaa** [comment]
reachable tests have nothing to do with netstat https://github.com/mizzy/specinfra/blob/master/lib/specinfra/command/base/host.rb#L17,

netcat is used on ubuntu (at least). netcat works together with serverspec don't work so smooth together. Also netcat is already preinstalled on rspec runner node.

PS. I would have added the port check since the begging if it had worked.

File: rake/spec/default/60-st2_all-services-ok_spec.rb (hunk 1 above)
Comment (dennybaa) [inline rake/spec/default/60-st2_all-services-ok_spec.rb]: not sure about the `expected` syntax, that I've provided the correct example, but anyway it's preferred to use.

Comment (armab) [inline rake/spec/default/60-st2_all-services-ok_spec.rb]: Good to know, will change :+1:

**Comment by armab** [comment]
@dennybaa please read messages more precisely.

You missed first commit: https://github.com/armab/st2-packages/commit/f86059e2fe570be56dca609c841c939945e176b7 with [`be_listening` (port)](https://github.com/mizzy/specinfra/blob/master/lib/specinfra/command/base/port.rb) checks.
And failing test which I mentioned: https://circleci.com/gh/armab/st2-packages/432

See the screenshot: 
xdt9m3u 1

Again, I'm blocked with `netstat` which doesn't exist in your Docker containers (need it on all distros, of course).

```sh
sudo lxc-attach -n "$(docker inspect --format '{{.Id}}' st2packages_wheezytest_1)" -- bash
root@8a951dadae64:/# netstat
bash: netstat: command not found

~$ sudo lxc-attach -n "$(docker inspect --format '{{.Id}}' st2packages_wheezybuild_1)" -- bash
root@27ae18530e96:/# netstat
bash: netstat: command not found
```

**Comment by armab** [comment]
Olala, it works
5lcpxnh 1

File: rake/spec/default/60-st2_all-services-ok_spec.rb [hunk 2]
```diff
@@ -63,4 +66,21 @@
       it { is_expected.to be_running }
     end
   end
+
+  describe 'st2auth', prompt_on_failure: true do
+    subject { port(9100) }
+    it { should be_listening }
```
Comment (dennybaa) [inline rake/spec/default/60-st2_all-services-ok_spec.rb]: same thing. use the same syntax, which is not should please.

Comment (armab) [inline rake/spec/default/60-st2_all-services-ok_spec.rb]: I guess this is kind of Ruby-madness around "beauty/ugly" :)

```ruby
describe 'st2api', prompt_on_failure: true do
  subject { port(9101) }
  it { is_expected.to be_listening }
end
```
To be sure, ^^ this is what you expect, right?

**Pull request merged**